test/gtest/ucm/malloc_hook.cc

```diff
@@ -716,14 +716,21 @@ class mmap_hooks {
             ASSERT_NE(-1, ret) << strerror(errno);
         }
 
-        /* 7. Unmap the buffer */
+        /* 7. Unmap the buffer using madvise */
+        {
+            madvise(buffer, size, MADV_DONTNEED);
```

Comment:
i don't think it's the right place to add this test
it should be in `void test_thread::test()`
allocate a new buffer and madvise(DONTNEED) it ..

Reply:
ok, shifted